app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -169,17 +165,25 @@ public void onItemRangeInserted(int positionStart, int itemCount) {
     private void signInAnonymously() {
         Toast.makeText(this, "Signing in...", Toast.LENGTH_SHORT).show();
         mAuth.signInAnonymously()
-                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
+                .addOnSuccessListener(this, new OnSuccessListener<AuthResult>() {
+                    @Override
+                    public void onSuccess(AuthResult result) {
+                        attachRecyclerViewAdapter();
+                    }
+                })
+                .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                     @Override
                     public void onComplete(@NonNull Task<AuthResult> task) {
-                        Log.d(TAG, "signInAnonymously:onComplete:" + task.isSuccessful());
                         if (task.isSuccessful()) {
-                            Toast.makeText(ChatActivity.this, "Signed In",
-                                    Toast.LENGTH_SHORT).show();
-                            attachRecyclerViewAdapter();
+                            Toast.makeText(getApplicationContext(),
```

Comment:
@samtstern I've extracted the toast logic into a static class to prevent activity leaks. As for the `attachRecyclerViewAdapter`, I moved that up to an onSuccess listener that is tied to the activity.